VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,924 @@ if ($PSEdition -eq 'Desktop') {
 
     if ("TrustAllCertificatePolicy" -as [type]) {} else {
         Add-Type @"
-	using System.Net;
+    using System.Net;
     using System.Security.Cryptography.X509Certificates;
     public class TrustAllCertificatePolicy : ICertificatePolicy {
         public TrustAllCertificatePolicy() {}
-		public bool CheckValidationResult(
+        public bool CheckValidationResult(
             ServicePoint sPoint, X509Certificate certificate,
             WebRequest wRequest, int certificateProblem) {
             return true;
         }
-	}
+    }
 "@
         [System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertificatePolicy
     }
 }
+
+
+#######################################################################################################################
+########################################################  FUNCTIONS  ##################################################
+
+
+Function Get-vCenterServerConnection {
+    <#
+        .SYNOPSIS
+        Get connection to vCenter Server via SDDC Manager using either domain or ESXi FQDN
+
+        .DESCRIPTION
+        The cmdlet connects to SDDC Manager using the -server, -user, and -password values
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that network connectivity and authentication is possible to vCenter Server
+        - Validates that the workload domain exists in the SDDC Manager inventory
+        - Connect to vCenter server and returns its details
+
+        .EXAMPLE
+        Get-vCenterServerConnection -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -esxiFqdn sfo01-m01-esx03.sfo.rainpole.io
+
+        .EXAMPLE 
+        Get-vCenterServerConnection -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -domain sfo-m01
```

Comment:
What’s the point for `esxiFqdn` here?

Reply:
either you pass domain or you pass esxiFqdn, not both. 

```
  .EXAMPLE
Get-vCenterServerConnection -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -esxiFqdn sfo01-m01-esx03.sfo.rainpole.io

 .EXAMPLE 
Get-vCenterServerConnection -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01
```